### DISCUSSION TITLE
Can't reach ollama => /ollama/api/api/tags

### DISCUSSION THREAD

**Discussion by benjaminantunes** [open-webui/open-webui#1136] [Q&A] [answered]
Hi,

I have a dumb trouble since I pulled newest update of open webui today (but i'm not sure the problem comes from this)

I can't reach Ollama because, inside the get request, there is two /api instead of one : 

INFO:     192.168.186.21:50946 - "GET /openai/api/models HTTP/1.1" 200 OK
INFO:     192.168.186.21:50946 - "GET /litellm/api/v1/models HTTP/1.1" 200 OK
INFO:     192.168.186.21:50946 - "GET /api/v1/chats/tags/all HTTP/1.1" 200 OK
INFO:     192.168.186.21:50956 - "GET /static/favicon.png HTTP/1.1" 200 OK
**INFO:     192.168.186.21:50946 - "GET /ollama/api/api/tags HTTP/1.1" 500 Internal Server Error**

In web browser console : 
Ollama: 404 Client Error: Not Found for url: http://localhost:11434/api/api/tags

However, I have inspected : 

ollamawebui/config.py : The base URL is fine ...
OLLAMA_API_BASE_URL = os.environ.get(
    "OLLAMA_API_BASE_URL", "http://localhost:11434/api"
)
OLLAMA_BASE_URL = os.environ.get("OLLAMA_BASE_URL", "http://localhost:11434")

I also looked inside ollamawebui/backend/apps/ollama/main.py : 

async def fetch_url(url):
    print("URL we are fetching : " + str(url))
    try:
        async with aiohttp.ClientSession() as session:
            async with session.get(url) as response:
                return await response.json()
    except Exception as e:
        # Handle connection error here
        print(f"Connection error: {e}")
        return None

The url is the right one !! : http://localhost:11434/api/tags

So, I don't understand where it is adding one /api ... 
Of course, http://localhost:11434/api/tags is working, but not http://localhost:11434/api/api/tags for clear reasons. 

The /api addition is happenning only for /ollama endpoints, in other cases it is fine. In my opinion, this should come from OLLAMA_BASE_URL, but looks like it is not coming from there ...

Getting stuck on such a dumb thing is quite annoying, thank you for your help !

**Comment by justinh-rahb** [comment, marked as answer]
Try updating again just to be sure you've got the lastest version, and change the `OLLAMA_API_BASE_URL=http://localhost:11434/api` to `OLLAMA_BASE_URL=http://localhost:11434` instead, note that the name of the variable has changed and we no longer put `/api` on the end of the URL.

### Docker (non-Compose method):
```bash
docker rm -f open-webui
docker pull ghcr.io/open-webui/open-webui:main
<the docker run command you used previously>
```

### Docker (Compose method):
```bash
docker compose down
docker pull ghcr.io/open-webui/open-webui:main
docker compose up -d
```

### Manual (direct method):
```bash
cd open-webui
git pull
npm i
npm run build
cd backend
pip3 install -r requirements.txt
bash start.sh
```

**Reply by benjaminantunes** [reply]
I just started again from zero, creating a new folder, and it is working. 
I can't figure out why it was working, then git pull, and then had this problem.

But starting from scratch solved the problem !
Thank you for your answer

**Comment by tjbck** [comment]
You might also want to delete your browser cache just in case!

**Reply by benjaminantunes** [reply]
Thank you, I tried but can't figure out what was the problem. Starting from scratch solved it.
